### PR TITLE
Revise the prose around phases.

### PR DESCRIPTION
**Pull request by munificent**
Nothing major here. I just tried to add a little motivation and context,
then did some copyediting to try to make things more concise and/or
clear. Let me know if you like any of the existing prose more. I'm happy
to iterate on it.

### CODE REVIEW

File: working/macros/feature-specification.md [hunk 1]
```diff
@@ -113,66 +113,116 @@ visible. For example, if two macros applied to two methods in the same class,
...
-In general, the introspection APIs are limited such that only things produced
-by previous phases can be introspected upon. This ensures we always give
-accurate and complete information in these APIs.
+```dart
+@serialize class Foo {
+  Bar bar;
+}
+
+class Bar {
+  memoize@ int three() => 3;
```
Comment (mateusfccp) [inline working/macros/feature-specification.md:126]: Shouldn't be `@memoize`, here?

```suggestion
  @memoize int three() => 3;
```

Comment (munificent) [reply working/macros/feature-specification.md]: Oops, yes. :)

File: working/macros/feature-specification.md [hunk 2]
```diff
@@ -113,66 +113,116 @@ visible. For example, if two macros applied to two methods in the same class,
...
+application of a Dart program looks like this:
 
-Since new types can be contributed to the program in this phase, very little
-reflective power is provided to these macros. You can see the names of types
-that are referenced in the declaration for instance, but you can't ask if they
-are a subtype of a known type, because we don't know how the types will resolve
-yet. Even a type which could be resolved to an existing type might not actually
-resolve to that type once macros are done (a new type could be introduced which
-shadows the original one).
+1.  For each library, ordered topologically by imports:
```
Comment (mateusfccp) [inline working/macros/feature-specification.md:149]: Is it intended that all the itens are `1.`?

Comment (jakemac53) [reply working/macros/feature-specification.md:149]: github at least renders these with actual sequential numbers when you do it like this (it makes writing the markdown a bit easier this way).

**Review by jakemac53 — APPROVED** [review]

File: working/macros/feature-specification.md (hunk 2 above)
Comment (munificent) [reply working/macros/feature-specification.md:149]: Yeah, Markdown automatically renumbers them.

Comment (mateusfccp) [reply working/macros/feature-specification.md:149]: Oh, I didn't know that.

**Pull request merged**